examples/step-16/step-16.cc

```diff
@@ -580,7 +580,7 @@ namespace Step16
 
     // With all this together, we can finally get about solving the linear
     // system in the usual way:
-    SolverControl            solver_control(1000, 1e-12);
+    SolverControl            solver_control(1000, 1e-6 * system_rhs.l2_norm());
     SolverCG<Vector<double>> solver(solver_control);
 
     solution = 0;
```

Comment:
Would it not be a better approach to always introduce the notion of relative and absolute tolerance instead?

We could specify something like std::max(1e-15, 1e-6 * system_rhs.l2_norm()) for example to prevent issues with machine precision.

Reply:
That only seems useful if the rhs could be zero.  And it's actively wrong if you're solving quantum problems in SI units where solution and right hand side might actually be on the order of 1e-16 :-)

Reply:
Ahh yeah you are right. I guess I am so used to solving Non-linear problems (e.g Newton's method) that I forgot about the case of linear problems!